File: VisaD.Application/Applications/Dtos/SpecialTrainingDto.cs
using VisaD.Application.Nomenclatures.Dtos;
using VisaD.Data.Nomenclatures;

namespace VisaD.Application.Applications.Dtos
{
	public class SpecialTrainingDto
	{
		public int Id { get; set; }

		public string Department { get; set; }

		public NomenclatureDto<TrainingType> Type { get; set; }

		public double Duration { get; set; }
		public NomenclatureDto<DurationType> DurationType { get; set; }
	}
}

File: WebApplication/src/modules/application/models/special-training.dto.ts
import { NomenclatureDto } from 'src/modules/nomenclature/common/models/nomenclature-dto';

export class SpecialTrainingDto {
  id: number;

  department: string;

  type: NomenclatureDto;
  duration: number;
  durationType: NomenclatureDto;
}
